api/src/services/sample-location-service.ts

```diff
@@ -64,13 +64,24 @@ export class SampleLocationService extends DBService {
    */
   async insertSampleLocations(sampleLocations: PostSampleLocations): Promise<SampleLocationRecord[]> {
     const methodService = new SampleMethodService(this.connection);
-
+    const determineName = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const nameKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'name' || key.toLowerCase() === 'label'
+      );
+      return nameKey && geometry.properties ? geometry.properties[nameKey].substring(0, 50) : '';
+    };
+    const determineDesc = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const descKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'desc' || key.toLowerCase() === 'description'
+      );
+      return descKey && geometry.properties ? geometry.properties[descKey].substring(0, 250) : '';
+    };
     // Create a sample location for each feature found
-    const promises = sampleLocations.survey_sample_sites.map((item, index) => {
+    const promises = sampleLocations.survey_sample_sites.map((item) => {
       const sampleLocation = {
         survey_id: sampleLocations.survey_id,
-        name: `Sample Site ${index + 1}`, // Business requirement to default the names to Sample Site # on creation
-        description: sampleLocations.description,
+        name: determineName(item), // Please note that additional logic for this also happens in insertSampleLocation below
```

Comment:
This comment isn't very helpful. What is the additional logic? And I don't think `insertSampleLocation` is below? (Could be wrong)

Reply:
I can make this comment more precise but insertSampleLocation is called on line 88.